### PR TITLE
Feature/ci cd

### PR DESCRIPTION
**Pull request by hermannleboss**
# Description

Prepare the CI CD for backend deployement

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Checklist:

* **Please check if the PR fulfills these requirements**

- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes/features)
- [x] Docs have been added / updated (for bug fixes / features)

### CODE REVIEW

File: backend/Dockerfile [hunk 1]
```diff
@@ -2,7 +2,7 @@
 FROM node:18-alpine as development  
 
 # Set the working directory inside the container
-WORKDIR /backend
+WORKDIR /app
```
Comment (dasarathg68) [inline backend/Dockerfile:5]: I have a doubt:
Why is the working directory set to /app when this is the dockerfile for the backend and commands below are executed relative to the backend?

Comment (hermannleboss) [reply backend/Dockerfile:5]: Workdir mean where docker will put the content inside his Virtual Machine
it's not related to the current directory. 

I often use /app so people remember that the WORKDIR is inside docker and not related to the project directory structure

Comment (dasarathg68) [reply backend/Dockerfile:5]: oh right. but don't you feel that it's confusing since our frontend codebase lies in /app?

Comment (hermannleboss) [reply backend/Dockerfile:5]: Yes i agree. 
I don't find a better name to descript it. 

What do you propose ?

Comment (dasarathg68) [reply backend/Dockerfile:5]: How about /docker_workdir or /docker_context or /docker_app ?

Comment (hermannleboss) [reply backend/Dockerfile:5]: I think /docker_workdir is good

**Review by hermannleboss — COMMENTED** [review]
LGTM

**Review by hermannleboss — COMMENTED** [review]
.

**Pull request merged**